src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5011,8 +5011,10 @@ void  MacroAssembler::decode_heap_oop(Register d, Register s) {
   verify_heapbase("MacroAssembler::decode_heap_oop: heap base corrupted?");
 #endif
   if (CompressedOops::base() == nullptr) {
-    if (CompressedOops::shift() != 0 || d != s) {
+    if (CompressedOops::shift() != 0) {
       lsl(d, s, CompressedOops::shift());
+    } else if (d != s) {
+      mov(d, s);
```

Comment:
I think `MacroAssembler::mov` already does the `d != s` optimization, but perhaps it's clearer to do it here as well.

Reply:
It is not very clear if you fine with this patch, @aph, or you want me to skip `d != s` check here. :)

Reply:
OK, no problem. Want to formally approve the PR?